src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -42,7 +42,8 @@ ss::future<> seq_writer::read_sync() {
 }
 
 ss::future<> seq_writer::wait_for(model::offset offset) {
-    return container().invoke_on(0, _smp_opts, [offset](seq_writer& seq) {
+    return container().invoke_on(shard, _smp_opts, [offset](seq_writer& seq) {
+        vassert(shard == ss::this_shard_id(), "This must run on shard 0");
```

Comment:
This doesn't really check that it's running on shard 0 though, right?  This just verifies that seastar did the right thing.

I guess you want `0 == ss::this_shard_id()` ?